### PR TITLE
Implements user repo

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR implements the user repo and adds integration tests for it. It changes the make command for triggering these tests.

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/golang/lib/repos/user.go [hunk 1]
```diff
@@ -16,21 +16,17 @@ type User interface {
...
 
 type userWriter interface {
 	// Creates inserts a new User with the specified fields.
 	Create(
 		ctx context.Context,
-		email string,
 		orgID string,
-		role string,
 		apiKey string,
-		auth0ID string,
```
Comment (eunice-chan) [inline src/golang/lib/repos/user.go:33]: Why did we make these changes?

File: src/golang/lib/repos/user.go [hunk 2]
```diff
@@ -16,21 +16,17 @@ type User interface {
 
 type userReader interface {
 	// GetByAPIKey returns the User with the API key apiKey.
+	// It returns a database.ErrNoRows if no rows are found.
 	GetByAPIKey(ctx context.Context, apiKey string, DB database.Database) (*models.User, error)
-
```
Comment (eunice-chan) [inline src/golang/lib/repos/user.go:20]: Why did we remove this?

File: src/golang/lib/repos/tests/seed.go [hunk 3]
```diff
@@ -1,6 +1,28 @@
...
+	"github.com/aqueducthq/aqueduct/lib/models"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// Defaults used for seeding database records
+	testOrgID = "aqueduct-test"
+)
+
+// seedUser creates count user records.
```
Comment (eunice-chan) [inline src/golang/lib/repos/tests/seed.go:13]: Nit: Put backticks around count to make it more explicit it is a variable.
e.g. // seedUser creates `count` user records.

**Review by eunice-chan — APPROVED** [review]

File: src/golang/lib/repos/user.go (hunk 2 above)
Comment (saurav-c) [reply src/golang/lib/repos/user.go:20]: `GetOrgAdmin` was unused and for now the open source doesn't have any admin users, it just has a single user.

File: src/golang/lib/repos/user.go (hunk 1 above)
Comment (saurav-c) [reply src/golang/lib/repos/user.go:33]: Similarly, the higher level call to this was just passing in an empty string for all these columns. We can add back these parameters if and when we have multiple users and start using these fields.

**Pull request merged**